File: 01.Python/1991.py
from typing import List

class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        totalSum = sum(nums)
        left = 0

        for i in range(len(nums)):
            right = totalSum - left - nums[i]

            if left == right:
                return i

            left += nums[i]
            
        
        return -1

File: 04.C/1991.c
int findMiddleIndex(int* nums, int numsSize) {
    int totalSum = 0;
    for (int i = 0; i < numsSize; i++) {
        totalSum += nums[i];
    }

    int left = 0;
    for (int i = 0; i < numsSize; i++) {
        int right = totalSum - left - nums[i];

        if (right == left) {
            return i;
        }

        left += nums[i];
    }

    return -1;
}